src/library/scala/Predef.scala

```diff
@@ -342,6 +342,49 @@ object Predef extends LowPriorityImplicits {
 
   // implicit classes -----------------------------------------------------
 
+  /** Adds `tap` method to every type.
+   *  `tap` applies `f` to the value for its side effects, and returns the original value.
+   *
+   *    {{{
+   *    val xs = List(1, 2, 3)
+   *               .tap(ys => println("debug " + ys.toString))
+   *    // xs == List(1, 2, 3)
+   *    }}}
+   *
+   *  @group implicit-classes-any */
+  implicit final class AnyTap[A](private val self: A) extends AnyVal {
```

Comment:
Why have two separate classes for these?

Also, any chance of turning these into a macro?  That would be a huge value-add because you wouldn't need to create the closure etc. in order to use this workflow.

Reply:
`String_+` isn't a macro; it's a synthetic method that the compiler injects into `String`.

That said, there are quite a few macros in stdlib, such as the `f""` string interpolator, `reify`, and quasiquotes. Their implementations are linked up in [`FastTrack`](https://github.com/scala/scala/blob/5687621cc117f79e63ac1a569a58a07c66ae6495/src/compiler/scala/tools/reflect/FastTrack.scala), which is presumably how you'd want to link these in.